assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -559,6 +561,54 @@ export const DEFAULT_NOTIFICATIONS = {
 			);
 		},
 	},
+	'web-data-stream-not-available-notification': {
+		Component: WebDataStreamNotAvailableNotification,
+		priority: 10,
+		areaSlug: NOTIFICATION_AREAS.BANNERS_ABOVE_NAV,
+		viewContexts: [ VIEW_CONTEXT_MAIN_DASHBOARD ],
+		isDismissible: false,
+		checkRequirements: async ( { select, resolveSelect }, viewContext ) => {
+			const viewOnly =
+				SITE_KIT_VIEW_ONLY_CONTEXTS.includes( viewContext );
+
+			await resolveSelect( CORE_MODULES ).getModules();
+
+			const ga4ModuleConnected = await select(
+				CORE_MODULES
+			).isModuleConnected( 'analytics-4' );
+
+			const hasGTMScope = await select( CORE_USER ).hasScope( GTM_SCOPE );
+			const loggedInUserID = await select( CORE_USER ).getID();
+			const ga4OwnerID = await select( MODULES_ANALYTICS_4 ).getOwnerID();
+
+			const isGA4ModuleOwner = () => {
+				// Bail early if we're in view-only dashboard or the GA4 module is not connected.
+				if ( viewOnly || ! ga4ModuleConnected ) {
+					return false;
+				}
+
+				if (
+					ga4OwnerID === undefined ||
+					loggedInUserID === undefined
+				) {
+					return undefined;
+				}
```

Comment:
The undefined condition won't exist as we will await resolution of all selectors before.